Core/src/IfcGeometryConverter/RepresentationConverter.h

```diff
@@ -510,6 +510,59 @@ namespace OpenInfraPlatform {
 						return;
 					}
 
+					// (3/*) IfcFace SUBTYPE OF IfcTopologicalRepresentationItem
+					// until now, this if-block isn't tested with an ifc-file
+					if (topo_item.isOfType<typename IfcEntityTypesT::IfcFace>()) {
+						// Carve polygon of the converted face
+						std::shared_ptr<carve::input::PolyhedronData> polygon(new carve::input::PolyhedronData());
+						// Contains polygon indices of vertices (x,y,z converted to string)
+						std::map<std::string, uint32_t> polygonIndices;
+						faceConverter->convertIfcFace(topo_item.as<typename IfcEntityTypesT::IfcFace>(), objectPlacement, polygon, polygonIndices);
+						itemData->open_or_closed_polyhedrons.push_back(polygon);
+						return;
+					}
+
+					// (4/*) IfcFaceBound SUBTYPE OF IfcTopologicalRepresentationItem
+					// until now, this if-block isn't tested with an ifc-file
+					if (topo_item.isOfType<typename IfcEntityTypesT::IfcFaceBound>()) {
+						// get vertices of the boundary
+						const std::vector<carve::geom::vector<3>> vertices = faceConverter->convertIfcFaceBound(
+							topo_item.as<typename IfcEntityTypesT::IfcFaceBound>(), objectPlacement);
+
+						// Carve polyline of the converted boundary
+						std::shared_ptr<carve::input::PolylineSetData> polyline_data(new carve::input::PolylineSetData());
+						// start a new polyline
+						polyline_data->beginPolyline();
+
+						// add all vertices to polyline
+						for (auto v : vertices) {

```

Comment:
```suggestion
						for (const auto& v : vertices) {
```

I'm not sure this would work, please try.

Reply:
If work == compiles: Done in commit 363c1cf.
However, I wasn't able to run this code block with an ifc file. Probably, I will create one for an unit test of `IfcTopologicalRepresentationItem`'s a bit later (especially to test the throw statements inside the called functions and the issue of an `IfcFace` with hole which I observed [above](https://github.com/tumcms/Open-Infra-Platform/pull/462#issuecomment-864908637)).